### ISSUE TITLE
Class.Type.Attributes is empty even when Attribute is applied

### ISSUE THREAD

**Issue by Yaevh** [TNG/ArchUnitNET#59]
Consider the following test (using `ArchUnitNET.xUnit`):

```
using static ArchUnitNET.Fluent.ArchRuleDefinition;

public class ArchitectureTests
{
    [Obsolete]
    public class ObsoleteClass { }

    
    private static readonly  Architecture _architecture =
            new ArchLoader().LoadAssemblies(typeof(ArchitectureTests).Assembly).Build();

    private static readonly IObjectProvider<IType> _obsoleteClasses =
            Classes().That().Are(typeof(ObsoleteClass))
                .As("obsolete classes");


    [Fact(DisplayName = "should be true and is true")]
    public void ObsoleteClass_should_have_ObsoleteAttribute__by_HaveAnyAttributes()
    {
        var archRule = Classes().That()
            .Are(_obsoleteClasses)
            .Should().HaveAnyAttributes(typeof(ObsoleteAttribute));

        ArchRuleAssert.CheckRule(_architecture, archRule);
    }

    [Fact(DisplayName = "should be true, but is false")]
    public void ObsoleteClass_should_have_ObsoleteAttribute__by_FollowCustomCondition()
    {
        var archRule = Classes().That()
            .Are(_obsoleteClasses)
            .Should().FollowCustomCondition(
                @class => @class.Type.Attributes.Any(),
                "ObsoleteClass should have ObsoleteAttribute",
                "ObsoleteClass should have ObsoleteAttribute");

        ArchRuleAssert.CheckRule(_architecture, archRule);
    }
}
```

In version 0.5.0 running under .NET 5, the first test passes as expected - but the second one fails, since `Class.Type.Attributes` list is empty, even though it should contain `ObsoleteAttribute`.

**Comment by Yaevh** [comment]
I've just doscovered that if I modify the failing test condition from `@class.Type.Attributes.Any()` to `@class.Attributes.Any()`, the test passes. This begs the question - what does `class.Type` represent? It would be helpful to provide some codedoc for `Type` property, as it can be confusing without it.

**Comment by brandhuf** [comment]
`class.Type` is mainly used during creation of the architecture. For writing tests you should only use `class.Attributes`. The `Type` property will probably be removed or made private in a future release.

**Comment by fgather** [comment]
Fix released in 0.10.0
